test/shared_test/lib_battery_dispatch_test.cpp

```diff
@@ -56,7 +58,7 @@ TEST_F(BatteryDispatchTest, DispatchAutoBTM_lib_battery_dispatch)
 	// TEST 1: Verify no grid charging since disallowed
 	dispatchAutoBTM->dispatch(0, 0, 0);
 	EXPECT_EQ(batteryPower->powerGridToBattery, 0);
-	EXPECT_EQ(batteryPower->powerBatteryDC, 0);
+	EXPECT_NEAR(batteryPower->powerBatteryDC, -39.42, 0.02);
```

Comment:
why did this answer change?

Reply:
this has to do with the battery's constraint checker not being able to enforce restrictions and targets properly. In the [constraint checker](https://github.com/NREL/ssc/blob/64c10e5295acbe58e4f75bb284931fc0ff10262c/shared/lib_battery_dispatch.cpp#L758) for this dispatch model, it'll first make sure the no grid charge constraint is enforced in dispatch_t::check_constraints line 761 >> line 161, then it'll move on to line 785 and try to enforce that the battery meets the target. However, the target wasn't changed during the process of checking that the grid charge constraint is met (in fact is never changed through the constraint checking) so then the battery power shoots back from 0, and we have -39.42 power charge.